cocos/2d/CCNode.cpp

```diff
@@ -1127,21 +1138,22 @@ void Node::insertChild(Node* child, int z)
     _transformUpdated = true;
     _reorderChildDirty = true;
     _children.pushBack(child);
-    child->_localZOrder = z;
+    child->_setLocalZOrder(z);
 }
 
 void Node::reorderChild(Node *child, int zOrder)
 {
     CCASSERT( child != nullptr, "Child must be non-nil");
     _reorderChildDirty = true;
-    child->_localZOrder = zOrder;
+    child->_updateOrderOfArrival();
+    child->_setLocalZOrder(zOrder);
 }
 
 void Node::sortAllChildren()
 {
     if (_reorderChildDirty)
     {
-        std::stable_sort(std::begin(_children), std::end(_children), nodeComparisonLess);
```

Comment:
Ah nevermind, I see it's all about going back to std::sort :D 
I should've read the original post instead of just the commit.